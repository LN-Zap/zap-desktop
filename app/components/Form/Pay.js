import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Isvg from 'react-inlinesvg'
import paperPlane from 'icons/paper_plane.svg'
import link from 'icons/link.svg'
import FaAngleDown from 'react-icons/lib/fa/angle-down'

import { btc } from 'utils'

import styles from './Pay.scss'

class Pay extends Component {
  componentDidUpdate(prevProps) {
    const {
      isOnchain,
      isLn,
      payform: { payInput },
      fetchInvoice
    } = this.props

    // If on-chain, focus on amount to let user know it's editable
    if (isOnchain) {
      this.amountInput.focus()
    }

    // If LN go retrieve invoice details
    if (prevProps.payform.payInput !== payInput && isLn) {
      fetchInvoice(payInput)
    }
  }

  render() {
    const {
      payform: { payInput, showErrors, invoice, showCurrencyFilters },
      nodes,
      ticker,

      isOnchain,
      isLn,
      currentAmount,
      usdAmount,
      payFormIsValid: { errors, isValid },
      currentCurrencyFilters,
      currencyName,

      setPayAmount,
      onPayAmountBlur,

      setPayInput,
      onPayInputBlur,

      setCurrencyFilters,

      onPaySubmit,

      setCurrency
    } = this.props

    const displayNodeName = pubkey => {
      const node = nodes.find(n => n.pub_key === pubkey)

      if (node && node.alias.length) {
        return node.alias
      }

      return pubkey ? pubkey.substring(0, 10) : ''
    }

    const onCurrencyFilterClick = currency => {
      if (!isLn) {
        // change the input amount
        setPayAmount(btc.convert(ticker.currency, currency, currentAmount))
      }

      setCurrency(currency)
      setCurrencyFilters(false)
    }

    return (
      <div className={styles.container}>
        <header className={styles.header}>
          <Isvg src={paperPlane} />
          <h1>Make Payment</h1>
        </header>

        <div className={styles.content}>
          <section className={styles.destination}>
            <div className={styles.top}>
              <label htmlFor="paymentRequest">Destination</label>
              <span className={`${styles.description} ${(isOnchain || isLn) && styles.active}`}>
                {isOnchain && (
                  <i>
                    <Isvg src={link} />
                    <span>On-Chain (~10 minutes)</span>
                  </i>
                )}
                {isLn && (
                  <i>
                    <span>
                      {displayNodeName(invoice.destination)} ({invoice.description})
                    </span>
                  </i>
                )}
              </span>
            </div>
            <div className={styles.bottom}>
              <textarea
                type="text"
                placeholder="Paste payment request or bitcoin address here"
                value={payInput}
                onChange={event => setPayInput(event.target.value)}
                onBlur={onPayInputBlur}
                id="paymentRequest"
                rows="4"
              />
              <section className={`${styles.errorMessage} ${showErrors.payInput && styles.active}`}>
                {showErrors.payInput && <span>{errors.payInput}</span>}
              </section>
            </div>
          </section>

          <section className={styles.amount}>
            <div className={styles.top}>
              <label htmlFor="amount">Amount</label>
              <span />
            </div>
            <div className={styles.bottom}>
              <input
                type="number"
                min="0"
                ref={input => {
                  this.amountInput = input
                }}
                size=""
                placeholder="0.00000000"
                value={currentAmount || ''}
                onChange={event => setPayAmount(event.target.value)}
                onBlur={onPayAmountBlur}
                id="amount"
                readOnly={isLn}
              />
              <div className={styles.currency}>
                <section
                  className={styles.currentCurrency}
                  onClick={() => setCurrencyFilters(!showCurrencyFilters)}
                >
                  <span>{currencyName}</span>
                  <span>
                    <FaAngleDown />
                  </span>
                </section>
                <ul className={showCurrencyFilters && styles.active}>
                  {currentCurrencyFilters.map(filter => (
                    <li key={filter.key} onClick={() => onCurrencyFilterClick(filter.key)}>
                      {filter.name}
                    </li>
                  ))}
                </ul>
              </div>
            </div>

            <div className={styles.usdAmount}>{`â‰ˆ ${usdAmount || 0} USD`}</div>

            <section
              className={`${styles.errorMessage} ${styles.amount} ${showErrors.amount &&
                styles.active}`}
            >
              {showErrors.amount && <span>{errors.amount}</span>}
            </section>
          </section>

          <section className={styles.submit}>
            <div className={`${styles.button} ${isValid && styles.active}`} onClick={onPaySubmit}>
              Pay
            </div>
          </section>
        </div>
      </div>
    )
  }
}

Pay.propTypes = {
  payform: PropTypes.shape({
    amount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    payInput: PropTypes.string.isRequired,
    invoice: PropTypes.object.isRequired,
    showErrors: PropTypes.object.isRequired
  }).isRequired,
  currencyName: PropTypes.string.isRequired,

  isOnchain: PropTypes.bool.isRequired,
  isLn: PropTypes.bool.isRequired,
  currentAmount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  usdAmount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  payFormIsValid: PropTypes.shape({
    errors: PropTypes.object,
    isValid: PropTypes.bool
  }).isRequired,

  setPayAmount: PropTypes.func.isRequired,
  onPayAmountBlur: PropTypes.func.isRequired,
  setPayInput: PropTypes.func.isRequired,
  onPayInputBlur: PropTypes.func.isRequired,
  fetchInvoice: PropTypes.func.isRequired,

  onPaySubmit: PropTypes.func.isRequired,
  setCurrencyFilters: PropTypes.func.isRequired,
  setCurrency: PropTypes.func.isRequired,

  ticker: PropTypes.object.isRequired,

  nodes: PropTypes.array.isRequired,
  currentCurrencyFilters: PropTypes.array.isRequired
}

export default Pay
