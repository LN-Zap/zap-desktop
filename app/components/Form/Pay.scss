@import '../../styles/variables.scss';

.container {
  padding: 0 40px;
  font-family: Roboto;
  margin: 0 auto;
  width: 500px;
}

.header {
  text-align: center;
  padding-bottom: 20px;
  color: $white;
  border-bottom: 1px solid $spaceborder;

  h1 {
    font-size: 22px;
    font-weight: 100;
    margin-top: 10px;
    letter-spacing: 1.5px;
  }
}

.content {
  margin-top: 40px;
  color: $white;

  .destination {
    margin-bottom: 10px;

    .description {
      font-size: 12px;
      line-height: 14px;
      padding: 0px 15px;
      min-height: 14px;

      &.active {
        background: #23252f;
        border-radius: 10px;
        min-height: 0;
      }
    }

    svg {
      width: 10px;
      height: 10px;
      line-height: 14px;
      margin-right: 5px;
    }
  }

  .amount .bottom {
    display: flex;
    flex-direction: row;
    align-items: center;

    input {
      font-size: 20px;
      max-width: 150px;
      border:1px solid #404040;
      border-radius: 4px;
      padding: 15px;
    }
  }

  .top {
    margin-bottom: 10px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    label {
      font-size: 14px;
    }
  }

  .bottom {
    input, textarea {
      background: transparent;
      outline: none;
      border: 1px solid #404040;
      color: $gold;
      -webkit-text-fill-color: $white;
      font-size: 12px;
      width: 100%;
      font-weight: 200;
      padding: 10px;
    }

    input::-webkit-input-placeholder, textarea::-webkit-input-placeholder {
      text-shadow: none;
      -webkit-text-fill-color: initial;
    }
  }

  .currency {
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-left: 12px;

    .currentCurrency {
      cursor: pointer;
      transition: 0.25s all;

      &:hover {
        opacity: 0.5;
      }

      span {
        font-size: 14px;
        
        &:nth-child(1) {
          font-weight: bold;
        }
      }

    }

    ul {
      visibility: hidden;
      position: absolute;
      top: 30px;

      &.active {
        visibility: visible;
      }

      li {
        padding: 8px 15px;
        background: #191919;
        cursor: pointer;
        transition: 0.25s hover;
        border-bottom: 1px solid #0f0f0f;

        &:hover {
          background: #0f0f0f;
        }
      }
    }
  }

  .usdAmount {
    margin-top: 10px;
    opacity: 0.5;
    font-size: 14px;
  }

  .submit {
    margin-top: 20px;
    text-align: center;

    .button {
      width: 235px;
      margin: 0 auto;
      padding: 20px 10px;
      background: #31343f;
      opacity: 0.5;
      cursor: pointer;
      transition: 0.25s all;

      &.active {
        background: $gold;
        opacity: 1.0;

        &:hover {
          background: darken($gold, 5%);
        }
      }
    }
  }
}

.errorMessage {
  color: $red;
  font-size: 12px;
  min-height: 20px;
  opacity: 0;
  transition: all 0.25s ease;

  &.amount {
    margin-top: 10px;
  }

  &.active {
    opacity: 1;
  }
}
