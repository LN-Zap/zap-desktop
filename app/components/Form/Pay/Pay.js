import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Isvg from 'react-inlinesvg'
import paperPlane from 'icons/paper_plane.svg'
import link from 'icons/link.svg'
import { MdKeyboardArrowDown } from 'react-icons/md'

import { btc } from 'lib/utils'
import AmountInput from 'components/AmountInput'

import { FormattedMessage, injectIntl } from 'react-intl'
import messages from './messages'

import styles from './Pay.scss'

class Pay extends Component {
  componentDidUpdate(prevProps) {
    const {
      isLn,
      payform: { payInput },
      fetchInvoice
    } = this.props

    // If LN go retrieve invoice details
    if (prevProps.payform.payInput !== payInput && isLn) {
      fetchInvoice(payInput)
    }
  }

  render() {
    const {
      payform: { payInput, showErrors, invoice, showCurrencyFilters },
      nodes,
      ticker,

      isOnchain,
      isLn,
      currentAmount,
      fiatAmount,
      payFormIsValid: { errors, isValid },
      currentCurrencyFilters,
      currencyName,

      setPayAmount,
      onPayAmountBlur,

      setPayInput,
      onPayInputBlur,

      setCurrencyFilters,

      onPaySubmit,

      setCurrency,
      intl
    } = this.props

    const displayNodeName = pubkey => {
      const node = nodes.find(n => n.pub_key === pubkey)

      if (node && node.alias.length) {
        return node.alias
      }

      return pubkey ? pubkey.substring(0, 10) : ''
    }

    const onCurrencyFilterClick = currency => {
      if (!isLn) {
        // change the input amount
        setPayAmount(btc.convert(ticker.currency, currency, currentAmount))
      }

      setCurrency(currency)
      setCurrencyFilters(false)
    }

    return (
      <div className={styles.container}>
        <header className={styles.header}>
          <Isvg src={paperPlane} />
          <h1>
            <FormattedMessage {...messages.title} />
          </h1>
        </header>

        <div className={styles.content}>
          <section className={styles.destination}>
            <div className={styles.top}>
              <label htmlFor="paymentRequest">
                <FormattedMessage {...messages.destination} />
              </label>
              <span
                className={`${styles.description} ${isOnchain || isLn ? styles.active : undefined}`}
              >
                {isOnchain && (
                  <i>
                    <Isvg src={link} />
                    <span>
                      <FormattedMessage {...messages.onchain_description} />
                    </span>
                  </i>
                )}
                {isLn && (
                  <i>
                    <span>
                      {displayNodeName(invoice.destination)} ({invoice.description})
                    </span>
                  </i>
                )}
              </span>
            </div>
            <div className={styles.bottom}>
              <textarea
                type="text"
                placeholder={intl.formatMessage({ ...messages.request_placeholder })}
                value={payInput}
                onChange={event => setPayInput(event.target.value)}
                onBlur={onPayInputBlur}
                id="paymentRequest"
                rows="4"
              />
              <section
                className={`${styles.errorMessage} ${
                  showErrors.payInput ? styles.active : undefined
                }`}
              >
                {showErrors.payInput && <span>{errors.payInput}</span>}
              </section>
            </div>
          </section>

          <section className={styles.amount}>
            <div className={styles.top}>
              <label htmlFor="amount">
                <FormattedMessage {...messages.amount} />
              </label>
              <span />
            </div>
            <div className={styles.bottom}>
              <AmountInput
                id="amount"
                amount={currentAmount}
                currency={ticker.currency}
                onChangeEvent={setPayAmount}
                onBlurEvent={onPayAmountBlur}
                readOnly={isLn}
              />
              <div className={styles.currency}>
                <section
                  className={styles.currentCurrency}
                  onClick={() => setCurrencyFilters(!showCurrencyFilters)}
                >
                  <span>{currencyName}</span>
                  <span>
                    <MdKeyboardArrowDown />
                  </span>
                </section>
                <ul className={showCurrencyFilters ? styles.active : undefined}>
                  {currentCurrencyFilters.map(filter => (
                    <li key={filter.key} onClick={() => onCurrencyFilterClick(filter.key)}>
                      {filter.name}
                    </li>
                  ))}
                </ul>
              </div>
            </div>

            <div className={styles.fiatAmount}>{`â‰ˆ ${fiatAmount || 0} ${ticker.fiatTicker}`}</div>

            <section
              className={`${styles.errorMessage} ${styles.amount} ${
                showErrors.amount ? styles.active : undefined
              }`}
            >
              {showErrors.amount && <span>{errors.amount}</span>}
            </section>
          </section>

          <section className={styles.submit}>
            <div
              className={`${styles.button} ${isValid ? styles.active : undefined}`}
              onClick={onPaySubmit}
            >
              <FormattedMessage {...messages.pay} />
            </div>
          </section>
        </div>
      </div>
    )
  }
}

Pay.propTypes = {
  payform: PropTypes.shape({
    amount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    payInput: PropTypes.string.isRequired,
    invoice: PropTypes.object.isRequired,
    showErrors: PropTypes.object.isRequired
  }).isRequired,
  currencyName: PropTypes.string.isRequired,

  isOnchain: PropTypes.bool.isRequired,
  isLn: PropTypes.bool.isRequired,
  currentAmount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  fiatAmount: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  payFormIsValid: PropTypes.shape({
    errors: PropTypes.object,
    isValid: PropTypes.bool
  }).isRequired,

  setPayAmount: PropTypes.func.isRequired,
  onPayAmountBlur: PropTypes.func.isRequired,
  setPayInput: PropTypes.func.isRequired,
  onPayInputBlur: PropTypes.func.isRequired,
  fetchInvoice: PropTypes.func.isRequired,

  onPaySubmit: PropTypes.func.isRequired,
  setCurrencyFilters: PropTypes.func.isRequired,
  setCurrency: PropTypes.func.isRequired,

  ticker: PropTypes.object.isRequired,

  nodes: PropTypes.array.isRequired,
  currentCurrencyFilters: PropTypes.array.isRequired
}

export default injectIntl(Pay)
