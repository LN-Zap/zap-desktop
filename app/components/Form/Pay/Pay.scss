@import 'styles/variables.scss';

.container {
  padding: 0 40px;
  margin: 0 auto;
  width: 500px;
}

.header {
  text-align: center;
  padding-bottom: 20px;
  color: var(--primaryText);
  border-bottom: 1px solid $spaceborder;

  h1 {
    font-size: 22px;
    font-weight: 100;
    margin-top: 10px;
    letter-spacing: 1.5px;
  }

  svg g {
    stroke: var(--primaryText);
  }
}

.content {
  margin-top: 40px;
  color: var(--primaryText);

  .destination {
    margin-bottom: 10px;

    .description {
      font-size: 12px;
      line-height: 14px;
      padding: 0 15px;
      min-height: 14px;

      &.active {
        background: var(--lightBackground);
        border-radius: 10px;
        min-height: 0;
      }
    }

    svg {
      width: 10px;
      height: 10px;
      line-height: 14px;
      margin-right: 5px;
    }
  }

  .amount .bottom {
    display: flex;
    flex-direction: row;
    align-items: center;

    input {
      font-size: 20px;
      max-width: 150px;
      border: 1px solid #404040;
      border-radius: 4px;
      padding: 15px;
    }
  }

  .top {
    margin-bottom: 10px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    label {
      font-size: 14px;
    }
  }

  .bottom {
    input,
    textarea {
      background: transparent;
      outline: none;
      border: 1px solid #404040;
      color: var(--lightningOrange);
      -webkit-text-fill-color: var(--primaryText);
      font-size: 12px;
      width: 100%;
      font-weight: 200;
      padding: 10px;
    }

    input::-webkit-input-placeholder,
    textarea::-webkit-input-placeholder {
      text-shadow: none;
      -webkit-text-fill-color: initial;
    }
  }

  .currency {
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-left: 12px;

    .currentCurrency {
      cursor: pointer;
      transition: 0.25s all;

      &:hover {
        opacity: 0.5;
      }

      span {
        font-size: 14px;

        &:nth-child(1) {
          font-weight: bold;
        }
      }
    }

    ul {
      visibility: hidden;
      position: absolute;
      top: 30px;
      z-index: z("form", "pay");

      &.active {
        visibility: visible;
      }

      li {
        padding: 8px 15px;
        background: var(--lightBackground);
        cursor: pointer;
        transition: 0.25s hover;

        &:hover {
          background: var(--darkestBackground);
        }
      }
    }
  }

  .fiatAmount {
    margin-top: 10px;
    opacity: 0.5;
    font-size: 14px;
  }

  .submit {
    margin-top: 20px;
    text-align: center;
  }
}

.errorMessage {
  color: var(--superRed);
  font-size: 12px;
  min-height: 20px;
  opacity: 0;
  transition: all 0.25s ease;

  &.amount {
    margin-top: 10px;
  }

  &.active {
    opacity: 1;
  }
}
