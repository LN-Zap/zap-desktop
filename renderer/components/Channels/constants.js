export const TRANSACTION_SPEED_SLOW = 'TRANSACTION_SPEED_SLOW'
export const TRANSACTION_SPEED_MEDIUM = 'TRANSACTION_SPEED_MEDIUM'
export const TRANSACTION_SPEED_FAST = 'TRANSACTION_SPEED_FAST'
export const CHANNEL_LIST_VIEW_MODE_SUMMARY = 'summary'
export const CHANNEL_LIST_VIEW_MODE_CARD = 'card'
export const CHANNEL_DATA_VIEW_MODE_BASIC = 'CHANNEL_DATA_VIEW_MODE_BASIC'
export const CHANNEL_DATA_VIEW_MODE_FULL = 'CHANNEL_DATA_VIEW_MODE_FULL'
