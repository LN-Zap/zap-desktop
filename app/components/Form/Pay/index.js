import Pay from './Pay'

export default Pay
