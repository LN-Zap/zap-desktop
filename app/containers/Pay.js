import { connect } from 'react-redux'
import { Pay } from 'components/Pay'
import { tickerSelectors, setCurrency, setFiatTicker } from 'reducers/ticker'
import { queryFees, queryRoutes } from 'reducers/pay'
import { infoSelectors } from 'reducers/info'
import { sendCoins } from 'reducers/transaction'
import { payInvoice } from 'reducers/payment'

const mapStateToProps = state => ({
  chain: state.info.chain,
  network: infoSelectors.testnetSelector(state) ? 'testnet' : 'mainnet',
  cryptoName: tickerSelectors.cryptoName(state),
  channelBalance: state.balance.channelBalance,
  currentTicker: tickerSelectors.currentTicker(state),
  cryptoCurrency: state.ticker.currency,
  cryptoCurrencyTicker: tickerSelectors.currencyName(state),
  cryptoCurrencies: tickerSelectors.currencyFilters(state),
  fiatCurrencies: state.ticker.fiatTickers,
  fiatCurrency: state.ticker.fiatTicker,
  initialPayReq: state.pay.payReq,
  isQueryingFees: state.pay.isQueryingFees,
  isQueryingRoutes: state.pay.isQueryingRoutes,
  nodes: state.network.nodes,
  onchainFees: state.pay.onchainFees,
  routes: state.pay.routes,
  walletBalance: state.balance.walletBalance
})

const mapDispatchToProps = {
  payInvoice,
  setCryptoCurrency: setCurrency,
  setFiatCurrency: setFiatTicker,
  sendCoins,
  queryFees,
  queryRoutes
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Pay)
