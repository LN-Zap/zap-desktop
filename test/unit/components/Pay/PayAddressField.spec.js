import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { PayAddressField } from 'components/Pay'
import { PAY_FORM_STEPS } from 'components/Pay/constants'

const props = {
  chain: 'Bitcoin',
  handlePayReqChange: () => {},
  intl: {},
  isLn: null,
  network: 'testnet',
  redirectPayReq: {},
}

describe('component.Pay.PayAddressField', () => {
  describe('current step is "address"', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<PayAddressField {...props} currentStep={PAY_FORM_STEPS.address} />)
      expect(toJSON(wrapper)).toMatchSnapshot()
    })
  })

  describe('current step is "summary"', () => {
    describe('and it is an LN transaction', () => {
      it('should render correctly', () => {
        const wrapper = shallow(
          <PayAddressField {...props} currentStep={PAY_FORM_STEPS.summary} isLn />
        )
        expect(toJSON(wrapper)).toMatchSnapshot()
      })
    })

    describe('and it is an on-chain transaction', () => {
      it('should render correctly', () => {
        const wrapper = shallow(
          <PayAddressField {...props} currentStep={PAY_FORM_STEPS.summary} isLn={false} />
        )
        expect(toJSON(wrapper)).toMatchSnapshot()
      })
    })
  })
})
