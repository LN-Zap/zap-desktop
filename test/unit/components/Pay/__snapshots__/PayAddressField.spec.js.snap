// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.Pay.PayAddressField current step is "address" should render correctly 1`] = `
<styled.div
  className="element-show"
>
  <Component
    context={
      PayAddressField {
        "context": Object {},
        "focusPayReqInput": [Function],
        "getPaymentRequestLabel": [Function],
        "payReqInput": Object {
          "current": null,
        },
        "props": Object {
          "chain": "Bitcoin",
          "currentStep": "address",
          "handlePayReqChange": [Function],
          "intl": Object {},
          "isLn": null,
          "network": "testnet",
          "redirectPayReq": Object {},
        },
        "refs": Object {},
        "setState": [Function],
        "state": null,
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": Object {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useResponder": [Function],
              "useState": [Function],
            },
            "_element": <PayAddressField
              chain="Bitcoin"
              currentStep="address"
              handlePayReqChange={[Function]}
              intl={Object {}}
              isLn={null}
              network="testnet"
              redirectPayReq={Object {}}
            />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_renderPhaseUpdates": null,
            "_rendered": <ForwardRef(styled.div)
              className="element-show"
            >
              <Unknown
                context={[Circular]}
                state="big"
              >
                [Function]
              </Unknown>
            </ForwardRef(styled.div)>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
      }
    }
    state="big"
  >
    <Component />
  </Component>
</styled.div>
`;

exports[`component.Pay.PayAddressField current step is "summary" and it is an LN transaction should render correctly 1`] = `
<styled.div
  className="element-hide"
>
  <Component
    context={
      PayAddressField {
        "context": Object {},
        "focusPayReqInput": [Function],
        "getPaymentRequestLabel": [Function],
        "payReqInput": Object {
          "current": null,
        },
        "props": Object {
          "chain": "Bitcoin",
          "currentStep": "summary",
          "handlePayReqChange": [Function],
          "intl": Object {},
          "isLn": true,
          "network": "testnet",
          "redirectPayReq": Object {},
        },
        "refs": Object {},
        "setState": [Function],
        "state": null,
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": Object {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useResponder": [Function],
              "useState": [Function],
            },
            "_element": <PayAddressField
              chain="Bitcoin"
              currentStep="summary"
              handlePayReqChange={[Function]}
              intl={Object {}}
              isLn={true}
              network="testnet"
              redirectPayReq={Object {}}
            />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_renderPhaseUpdates": null,
            "_rendered": <ForwardRef(styled.div)
              className="element-hide"
            >
              <Unknown
                context={[Circular]}
                state="big"
              >
                [Function]
              </Unknown>
            </ForwardRef(styled.div)>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
      }
    }
    state="big"
  >
    <Component />
  </Component>
</styled.div>
`;

exports[`component.Pay.PayAddressField current step is "summary" and it is an on-chain transaction should render correctly 1`] = `
<styled.div
  className="element-hide"
>
  <Component
    context={
      PayAddressField {
        "context": Object {},
        "focusPayReqInput": [Function],
        "getPaymentRequestLabel": [Function],
        "payReqInput": Object {
          "current": null,
        },
        "props": Object {
          "chain": "Bitcoin",
          "currentStep": "summary",
          "handlePayReqChange": [Function],
          "intl": Object {},
          "isLn": false,
          "network": "testnet",
          "redirectPayReq": Object {},
        },
        "refs": Object {},
        "setState": [Function],
        "state": null,
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": Object {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useResponder": [Function],
              "useState": [Function],
            },
            "_element": <PayAddressField
              chain="Bitcoin"
              currentStep="summary"
              handlePayReqChange={[Function]}
              intl={Object {}}
              isLn={false}
              network="testnet"
              redirectPayReq={Object {}}
            />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_renderPhaseUpdates": null,
            "_rendered": <ForwardRef(styled.div)
              className="element-hide"
            >
              <Unknown
                context={[Circular]}
                state="small"
              >
                [Function]
              </Unknown>
            </ForwardRef(styled.div)>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
      }
    }
    state="small"
  >
    <Component />
  </Component>
</styled.div>
`;
